iocore/hostdb/P_RefCountCache.h

```diff
@@ -0,0 +1,896 @@
+/** @file
+
+  A cache (with map-esque interface) for RefCountObjs
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+#ifndef _P_RefCountCache_h_
+#define _P_RefCountCache_h_
+
+#include "I_EventSystem.h"
+#include "P_EventSystem.h" // TODO: less? just need ET_TASK
+
+#include <ts/Map.h>
+#include "ts/PriorityQueue.h"
+
+#include <ts/List.h>
+#include <ts/ink_hrtime.h>
+
+#include "ts/Vec.h"
+#include "ts/I_Version.h"
+#include <unistd.h>
+
+#define REFCOUNT_CACHE_EVENT_SYNC REFCOUNT_CACHE_EVENT_EVENTS_START
+
+#define REFCOUNTCACHE_MAGIC_NUMBER 0x0BAD2D9
+#define REFCOUNTCACHE_MAJOR_VERSION 1
+#define REFCOUNTCACHE_MINOR_VERSION 0
+
+// Stats
+enum RefCountCache_Stats {
+  refcountcache_current_items_stat,        // current number of items
+  refcountcache_current_size_stat,         // current size of cache
+  refcountcache_total_inserts_stat,        // total items inserted
+  refcountcache_total_failed_inserts_stat, // total items unable to insert
+  refcountcache_total_lookups_stat,        // total get() calls
+  refcountcache_total_hits_stat,           // total hits
+
+  // Persistence metrics
+  refcountcache_last_sync_time,   // seconds since epoch of last successful sync
+  refcountcache_last_total_items, // number of items sync last time
+  refcountcache_last_total_size,  // total size at last sync
+
+  RefCountCache_Stat_Count
+};
+
+struct RefCountCacheItemMeta {
+  uint64_t key;
+  unsigned int size;
+  ink_time_t expiry_time; // expire time as seconds since epoch
+  RefCountCacheItemMeta(uint64_t key, unsigned int size, int expire_time = -1) : key(key), size(size), expiry_time(expire_time) {}
```

Comment:
In some places you call this ``expire_time`` and in others you call it ``expiry_time``. Let's consistently use the latter.